bsip-0081.md

```diff
@@ -28,23 +28,28 @@ them to attract trading activities.
 
 There is already a flag `market_fee_percent` in asset options.
 
-Add a new flag `taker_fee_percent` into asset options, assign
-`market_fee_percent` to `taker_fee_percentage` at the consensus
-upgrade time for existing assets. The new flag can only be set
+Add a new optional flag `taker_fee_percent` into asset options, 
+which can only be set
 or updated by asset owners after the consensus upgrade.
+The valid range of the new flag is `[0%, 100%]`.
 
 Before the consensus upgrade, when an order buying that asset
 got filled, the amount `bought_amount * market_fee_percent`
-will go to the asset's accumulated fees.
+(capped by `max_market_fee`)
+will enter the asset's market fee sharing process.
 
 After the consensus upgrade, when an order buying the asset
 got filled,
-* if the order is a *maker*, the amount
-  `bought_amount * market_fee_percent` will go to the
-  asset's accumulated fees;
-* otherwise (if the order is a *taker*), the amount
-  `bought_amount * taker_fee_percent` will go to the
-  asset's accumulated fees.
+* if the order is a *maker*, or the `taker_fee_percent` flag
```

Comment:
"..., or the `taker_fee_percent` flag is not set in the asset ..."

Is this extra condition necessary? I think "... if the order is a *maker* ..." is already precise and whether the `taker_fee_percent` is set does not affect this anymore.

You could add in the first sentences
 - "Before the consensus upgrade, when market fee is enabled and an order buying that asset got filled ..."
 - "After the consensus upgrade, when market fee is enabled and order buying the asset got filled, ..." 
to clarify that both fees will only be collected if market fee is enabled.

Reply:
I'm not sure if you've understood the logic. I guess @MichelSantos has.
It's just how we describe it, as below:
```
if A
  do action 1
else if B
  do action 1
else
  do action 2
```
or
```
if A or B
  do action 1
else
  do action 2
```

Reply:
We're not talking about `market_fee_percent` here but `taker_fee_percent`, so your suggestion doesn't make sense.

Reply:
> You could add in the first sentences
> 
>     * "Before the consensus upgrade, when market fee is enabled and an order buying that asset got filled ..."
> 
>     * "After the consensus upgrade, when market fee is enabled and order buying the asset got filled, ..."
>       to clarify that both fees will only be collected if market fee is enabled.

This phrasing is possible too, and is technically correct.  The distinction being made in the current text is between the maker and taker fee, and it presumes that the the market fee is enabled.

Reply:
Thanks. I think I misread the or in the first if.